pandas/tests/plotting/test_series.py

```diff
@@ -569,7 +569,11 @@ def test_kde_missing_vals(self):
         _skip_if_no_scipy_gaussian_kde()
         s = Series(np.random.uniform(size=50))
         s[0] = np.nan
-        _check_plot_works(s.plot.kde)
+        axes = _check_plot_works(s.plot.kde)
+        # check if the values have any missing values
+        # GH14821
+        self.assertTrue(any(~np.isnan(axes.lines[0]._xorig)),
```

Comment:
Can you replace the `._xorig` with `.get_xdata()`? Better to not use matplotlib's internal APIs.